### ISSUE TITLE
install problems with conda gcc

### ISSUE THREAD

**Issue by katiebreivik** [COSMIC-PopSynth/COSMIC#128]
It looks like the conda gcc doesn't play nicely anymore after a mac upgrade to mojave. I have verified that the install works if you have the developer tools gcc version, and jeff has it working with a homebrew gcc. 

Here is an example of someone else running into this problem: https://github.com/openai/mujoco-py/issues/331

**Comment by michaelzevin** [comment]
Totally fixed!

**Comment by ReneKat** [comment]
@michaelzevin 
I'm having this issue myself and the `conda update gcc` doesn't fix it. Any guidance would be appreciated. Thank you!

I am running OSX Mojave 10.14.6 and have just updated Xcode to 11.3.1 trying to solve this gcc compilation error. 

`pip install git+https://github.com/danielfrg/tsne.git`

```
Collecting git+https://github.com/danielfrg/tsne.git
  Cloning https://github.com/danielfrg/tsne.git to /private/var/folders/5s/r0xdrzyj2dl_43mj6jh0cmzc0000gp/T/pip-req-build-wu_d_54h
  Running command git clone -q https://github.com/danielfrg/tsne.git /private/var/folders/5s/r0xdrzyj2dl_43mj6jh0cmzc0000gp/T/pip-req-build-wu_d_54h
Requirement already satisfied: Cython>=0.19.1 in /Users/Owl/miniconda3/envs/autometa/lib/python3.8/site-packages (from tsne==0.1.8) (0.29.15)
Requirement already satisfied: numpy>=1.7.1 in /Users/Owl/miniconda3/envs/autometa/lib/python3.8/site-packages (from tsne==0.1.8) (1.18.1)
Requirement already satisfied: scipy>=0.12.0 in /Users/Owl/miniconda3/envs/autometa/lib/python3.8/site-packages (from tsne==0.1.8) (1.4.1)
Building wheels for collected packages: tsne
  Building wheel for tsne (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /Users/Owl/miniconda3/envs/autometa/bin/python3.8 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/5s/r0xdrzyj2dl_43mj6jh0cmzc0000gp/T/pip-req-build-wu_d_54h/setup.py'"'"'; __file__='"'"'/private/var/folders/5s/r0xdrzyj2dl_43mj6jh0cmzc0000gp/T/pip-req-build-wu_d_54h/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/5s/r0xdrzyj2dl_43mj6jh0cmzc0000gp/T/pip-wheel-3xolta_u
       cwd: /private/var/folders/5s/r0xdrzyj2dl_43mj6jh0cmzc0000gp/T/pip-req-build-wu_d_54h/
  Complete output (29 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-10.9-x86_64-3.8
  creating build/lib.macosx-10.9-x86_64-3.8/tsne
  copying tsne/_version.py -> build/lib.macosx-10.9-x86_64-3.8/tsne
  copying tsne/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/tsne
  creating build/lib.macosx-10.9-x86_64-3.8/tsne/tests
  copying tsne/tests/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/tsne/tests
  copying tsne/tests/test_seed.py -> build/lib.macosx-10.9-x86_64-3.8/tsne/tests
  copying tsne/tests/test_iris.py -> build/lib.macosx-10.9-x86_64-3.8/tsne/tests
  UPDATING build/lib.macosx-10.9-x86_64-3.8/tsne/_version.py
  set build/lib.macosx-10.9-x86_64-3.8/tsne/_version.py to '0.1.8'
  running build_ext
  building 'bh_sne' extension
  creating build/temp.macosx-10.9-x86_64-3.8
  creating build/temp.macosx-10.9-x86_64-3.8/tsne
  creating build/temp.macosx-10.9-x86_64-3.8/tsne/bh_sne_src
  gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/Owl/miniconda3/envs/autometa/include -arch x86_64 -I/Users/Owl/miniconda3/envs/autometa/include -arch x86_64 -I/Users/Owl/miniconda3/envs/autometa/lib/python3.8/site-packages/numpy/core/include -Itsne/bh_sne_src/ -I/Users/Owl/miniconda3/envs/autometa/include/python3.8 -c tsne/bh_sne.cpp -o build/temp.macosx-10.9-x86_64-3.8/tsne/bh_sne.o -I/System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/Headers
  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++ [enabled by default]
  In file included from /Users/Owl/miniconda3/pkgs/gcc-4.8.2-21/lib/gcc/x86_64-apple-darwin13.0.0/4.8.2/install-tools/include/limits.h:34:0,
                   from /Users/Owl/miniconda3/envs/autometa/include/python3.8/Python.h:11,
                   from tsne/bh_sne.cpp:36:
  /Users/Owl/miniconda3/envs/autometa/lib/gcc/x86_64-apple-darwin11.4.2/4.8.5/include-fixed/syslimits.h:7:25: fatal error: limits.h: No such file or directory
   #include_next <limits.h>
                           ^
  compilation terminated.
  error: command 'gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for tsne
  Running setup.py clean for tsne
Failed to build tsne
Installing collected packages: tsne
    Running setup.py install for tsne ... error
    ERROR: Command errored out with exit status 1:
     command: /Users/Owl/miniconda3/envs/autometa/bin/python3.8 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/5s/r0xdrzyj2dl_43mj6jh0cmzc0000gp/T/pip-req-build-wu_d_54h/setup.py'"'"'; __file__='"'"'/private/var/folders/5s/r0xdrzyj2dl_43mj6jh0cmzc0000gp/T/pip-req-build-wu_d_54h/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/5s/r0xdrzyj2dl_43mj6jh0cmzc0000gp/T/pip-record-5uzqp_sm/install-record.txt --single-version-externally-managed --compile --install-headers /Users/Owl/miniconda3/envs/autometa/include/python3.8/tsne
         cwd: /private/var/folders/5s/r0xdrzyj2dl_43mj6jh0cmzc0000gp/T/pip-req-build-wu_d_54h/
    Complete output (29 lines):
    running install
    running build
    running build_py
    creating build
    creating build/lib.macosx-10.9-x86_64-3.8
    creating build/lib.macosx-10.9-x86_64-3.8/tsne
    copying tsne/_version.py -> build/lib.macosx-10.9-x86_64-3.8/tsne
    copying tsne/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/tsne
    creating build/lib.macosx-10.9-x86_64-3.8/tsne/tests
    copying tsne/tests/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/tsne/tests
    copying tsne/tests/test_seed.py -> build/lib.macosx-10.9-x86_64-3.8/tsne/tests
    copying tsne/tests/test_iris.py -> build/lib.macosx-10.9-x86_64-3.8/tsne/tests
    UPDATING build/lib.macosx-10.9-x86_64-3.8/tsne/_version.py
    set build/lib.macosx-10.9-x86_64-3.8/tsne/_version.py to '0.1.8'
    running build_ext
    building 'bh_sne' extension
    creating build/temp.macosx-10.9-x86_64-3.8
    creating build/temp.macosx-10.9-x86_64-3.8/tsne
    creating build/temp.macosx-10.9-x86_64-3.8/tsne/bh_sne_src
    gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/Owl/miniconda3/envs/autometa/include -arch x86_64 -I/Users/Owl/miniconda3/envs/autometa/include -arch x86_64 -I/Users/Owl/miniconda3/envs/autometa/lib/python3.8/site-packages/numpy/core/include -Itsne/bh_sne_src/ -I/Users/Owl/miniconda3/envs/autometa/include/python3.8 -c tsne/bh_sne.cpp -o build/temp.macosx-10.9-x86_64-3.8/tsne/bh_sne.o -I/System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/Headers
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++ [enabled by default]
    In file included from /Users/Owl/miniconda3/pkgs/gcc-4.8.2-21/lib/gcc/x86_64-apple-darwin13.0.0/4.8.2/install-tools/include/limits.h:34:0,
                     from /Users/Owl/miniconda3/envs/autometa/include/python3.8/Python.h:11,
                     from tsne/bh_sne.cpp:36:
    /Users/Owl/miniconda3/envs/autometa/lib/gcc/x86_64-apple-darwin11.4.2/4.8.5/include-fixed/syslimits.h:7:25: fatal error: limits.h: No such file or directory
     #include_next <limits.h>
                             ^
    compilation terminated.
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /Users/Owl/miniconda3/envs/autometa/bin/python3.8 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/5s/r0xdrzyj2dl_43mj6jh0cmzc0000gp/T/pip-req-build-wu_d_54h/setup.py'"'"'; __file__='"'"'/private/var/folders/5s/r0xdrzyj2dl_43mj6jh0cmzc0000gp/T/pip-req-build-wu_d_54h/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/5s/r0xdrzyj2dl_43mj6jh0cmzc0000gp/T/pip-record-5uzqp_sm/install-record.txt --single-version-externally-managed --compile --install-headers /Users/Owl/miniconda3/envs/autometa/include/python3.8/tsne Check the logs for full command output.
```


`$ which gcc`
/Users/Owl/miniconda3/envs/autometa/bin/gcc
`conda uninstall gcc`
...
The following packages will be REMOVED:

  cloog-0.18.0-10
  gcc-4.8.5-8
  gmp-5.1.2-6
  isl-0.12.2-1
  mpc-1.1.0-4
  mpfr-3.1.5-0
...

Opened new terminal and activated my environment
`$ which gcc`
/usr/bin/gcc

`pip install git+https://github.com/danielfrg/tsne.git`

```
Collecting git+https://github.com/danielfrg/tsne.git
  Cloning https://github.com/danielfrg/tsne.git to /private/var/folders/5s/r0xdrzyj2dl_43mj6jh0cmzc0000gp/T/pip-req-build-wpljhqig
  Running command git clone -q https://github.com/danielfrg/tsne.git /private/var/folders/5s/r0xdrzyj2dl_43mj6jh0cmzc0000gp/T/pip-req-build-wpljhqig
    ERROR: Command errored out with exit status 1:
     command: /Users/Owl/miniconda3/envs/autometa/bin/python3.8 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/5s/r0xdrzyj2dl_43mj6jh0cmzc0000gp/T/pip-req-build-wpljhqig/setup.py'"'"'; __file__='"'"'/private/var/folders/5s/r0xdrzyj2dl_43mj6jh0cmzc0000gp/T/pip-req-build-wpljhqig/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/5s/r0xdrzyj2dl_43mj6jh0cmzc0000gp/T/pip-req-build-wpljhqig/pip-egg-info
         cwd: /private/var/folders/5s/r0xdrzyj2dl_43mj6jh0cmzc0000gp/T/pip-req-build-wpljhqig/
    Complete output (49 lines):
    Traceback (most recent call last):
      File "/Users/Owl/miniconda3/envs/autometa/lib/python3.8/site-packages/numpy/core/__init__.py", line 24, in <module>
        from . import multiarray
      File "/Users/Owl/miniconda3/envs/autometa/lib/python3.8/site-packages/numpy/core/multiarray.py", line 14, in <module>
        from . import overrides
      File "/Users/Owl/miniconda3/envs/autometa/lib/python3.8/site-packages/numpy/core/overrides.py", line 7, in <module>
        from numpy.core._multiarray_umath import (
    ImportError: dlopen(/Users/Owl/miniconda3/envs/autometa/lib/python3.8/site-packages/numpy/core/_multiarray_umath.cpython-38-darwin.so, 2): Library not loaded: @rpath/libquadmath.0.dylib
      Referenced from: /Users/Owl/miniconda3/envs/autometa/lib/libcblas.3.dylib
      Reason: image not found
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/5s/r0xdrzyj2dl_43mj6jh0cmzc0000gp/T/pip-req-build-wpljhqig/setup.py", line 17, in <module>
        import numpy
      File "/Users/Owl/miniconda3/envs/autometa/lib/python3.8/site-packages/numpy/__init__.py", line 142, in <module>
        from . import core
      File "/Users/Owl/miniconda3/envs/autometa/lib/python3.8/site-packages/numpy/core/__init__.py", line 54, in <module>
        raise ImportError(msg)
    ImportError:
    
    IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!
    
    Importing the numpy c-extensions failed.
    - Try uninstalling and reinstalling numpy.
    - If you have already done that, then:
      1. Check that you expected to use Python3.8 from "/Users/Owl/miniconda3/envs/autometa/bin/python3.8",
         and that you have no directories in your PATH or PYTHONPATH that can
         interfere with the Python and numpy version "1.18.1" you're trying to use.
      2. If (1) looks fine, you can open a new issue at
         https://github.com/numpy/numpy/issues.  Please include details on:
         - how you installed Python
         - how you installed numpy
         - your operating system
         - whether or not you have multiple versions of Python installed
         - if you built from source, your compiler versions and ideally a build log
    
    - If you're working with a numpy git repository, try `git clean -xdf`
      (removes all files not under version control) and rebuild numpy.
    
    Note: this error has many possible causes, so please don't comment on
    an existing issue about this - open a new one instead.
    
    Original error was: dlopen(/Users/Owl/miniconda3/envs/autometa/lib/python3.8/site-packages/numpy/core/_multiarray_umath.cpython-38-darwin.so, 2): Library not loaded: @rpath/libquadmath.0.dylib
      Referenced from: /Users/Owl/miniconda3/envs/autometa/lib/libcblas.3.dylib
      Reason: image not found
    
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

**Comment by michaelzevin** [comment]
@scottcoughlin2014 would be the expert on how to fix these types of issues. Scotty, any thoughts?
